board/safety/safety_toyota.h

```diff
@@ -199,7 +207,40 @@ static int toyota_tx_hook(CANPacket_t *to_send, bool longitudinal_allowed) {
       int desired_torque = (GET_BYTE(to_send, 1) << 8) | GET_BYTE(to_send, 2);
       desired_torque = to_signed(desired_torque, 16);
       bool steer_req = GET_BIT(to_send, 0U) != 0U;
-      if (steer_torque_cmd_checks(desired_torque, steer_req, TOYOTA_STEERING_LIMITS)) {
+
+      // toyota overrides the steer request bit check
+      bool violation = steer_torque_cmd_checks(desired_torque, -1, TOYOTA_STEERING_LIMITS);
+
+      uint32_t ts = microsecond_timer_get();
+
+      // allow setting STEER_REQUEST bit low for a single frame to prevent EPS faults
+      bool steer_req_mismatch = (desired_torque != 0) && !steer_req;
+      if (!steer_req_mismatch) {
+        toyota_valid_steering_frame_count = MIN(toyota_valid_steering_frame_count + 1U, 255U);
+      } else {
+        // disallow torque cut if not enough recent matching steer_req messages
+        if (toyota_valid_steering_frame_count < (TOYOTA_MIN_VALID_STEERING_FRAMES - 1U)) {
+          violation = 1;
+        }
+        // or we've cut torque too recently in time
+        if ((ts - toyota_ts_last_steer_mismatch) < TOYOTA_RT_STEER_MISMATCH) {
+          violation = 1;
+        }
+
+        toyota_valid_steering_frame_count = 0U;
+        toyota_ts_last_steer_mismatch = ts;
+      }
```

Comment:
@adeebshihadeh want to make these common checks and add to the `SteeringLimits` struct? Probably a lot will be shared with Hyundai